controllers/v1/solutions.js

```diff
@@ -16,64 +16,1066 @@ module.exports = class  Solutions extends Abstract{
+    "status": 200,
+    "result": {
+        "_id": "6006a94d67f675771573226d"
+    }
+  }
+*/
+
+  /**
+ * Create solution.
+ * @method
+ * @name create
+ * @param {Object} req - requested data.
+ * req.body : {
+    "name": "create solution",
+    "programExternalId": "MAHARASTHA-AUTO-TARGETING",
+    "resourceType": [],
+    "language": [],
+    "keywords": [],
+    "concepts": [],
+    "themes": [],
+    "flattenedThemes": [],
+    "entities": [
+        "5beaa888af0065f0e0a10515",
+        "5fd098e2e049735a86b748ac"
+    ],
+    "registry": [],
+    "isRubricDriven": false,
+    "enableQuestionReadOut": false,
+    "allowMultipleAssessemts": false,
+    "isDeleted": false,
+    "entityType": "school",
+    "type": "improvementProject",
+    "subType": "improvementProject",
+    "isReusable": false,
+    "externalId": "01c04166-a65e-4e92-a87b-a9e4194e771lll",
+    "minNoOfSubmissionsRequired": 2,
+    "scope": {
+        "entityType": "block",
+        "entityTypeId": "5f32d8228e0dc8312404056e",
+        "entities": [
+            "5fd1b52ab53a6416aaeefc80",
+            "5fd098e2e049735a86b748ac",
+            "5fd1b52ab53a6416aaeefc83",
+            "5fd1b52ab53a6416aaeefb20"
+        ],
+        "roles": "BEO"
+    }
+  }
+
+ * @param {String} 
+ * @returns {JSON} Created solution data.
+ */
+
+  async create(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let solutionData = await solutionsHelper.createSolution(
+          req.userDetails.userToken,
+          req.body,
+          true                                    //this is true for when its called via API calls
+        );
+
+        return resolve(solutionData);
+
+      } catch (error) {
+        return reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        });
+      }
+    });
+  } 
+
+
+    /**
+* @api {post} /project/v1/solutions/update/:solutionId Update solution
+* @apiVersion 1.0.0
+* @apiName Update solution
+* @apiGroup Solutions
+* @apiParamExample {json} Request-Body:
+* {
+  "programExternalId" : "AMAN_TEST_123-1607937244986",
+  "entityType" : "school",
+  "externalId" : "IMPROVEMENT-PROJECT-TEST-SOLUTION",
+  "name" : "Improvement project test solution",
+  "description" : "Improvement project test solution"
+}
+* @apiHeader {String} internal-access-token internal access token  
+* @apiHeader {String} X-authenticated-user-token Authenticity token
+* @apiSampleRequest /project/v1/solutions/update/6006a94d67f675771573226d
+* @apiUse successBody
+* @apiUse errorBody
+* @apiParamExample {json} Response:
+* {
+  "message": "Solution updated successfully",
+  "status": 200,
+  "result": {
+      "_id": "6006a94d67f675771573226d"
+  }
+  }
+*/
+
+
+   /**
+   * Update solution.
+   * @method
+   * @name update
+   * @param {Object} req - requested data.
```

Comment:
@Prajwal17Tunerlabs  here you do not need to add the req body

Reply:
why should i not add req body ?